src/canvas.tsx

```diff
@@ -1,18 +1,93 @@
 import * as THREE from 'three'
-import React, { useRef, useEffect, useMemo, useState, useCallback, useContext } from 'react'
+import * as React from 'react'
+import { useRef, useEffect, useMemo, useState, useCallback } from 'react'
 import ResizeObserver from 'resize-observer-polyfill'
 import { invalidate, applyProps, render, unmountComponentAtNode } from './reconciler'
 
-export const stateContext = React.createContext()
+export type CanvasContext = {
+  canvas?: React.MutableRefObject<any>
+  subscribers: Array<Function>
+  frames: 0
```

Comment:
Should these two be numbers?

Reply:
subscribers is an array of callbacks. frames is an int.

Reply:
Yeah, I meant to comment on the line below, as the two 0s will be Numeric Literal Types and thus only allow actual 0 as value.